src/backend/commands/matview.c

```diff
@@ -3451,24 +3453,39 @@ build_query_for_apply_old_delta(const char *matviewname, const char *deltaname_o
 					updt_returning,
 					matviewname,
 					select_for_recalc);
-	return querybuf.data;
+
+	if (SPI_exec(querybuf.data, 0) != SPI_OK_SELECT)
+		elog(ERROR, "SPI_exec failed: %s", querybuf.data);
+
+	if (minmax_list)
+	{
+		if (tuptable_recalc)
+			*tuptable_recalc = SPI_tuptable;
+		return SPI_processed;
+	}
+	else
+	{
+		if (tuptable_recalc)
+			*tuptable_recalc = NULL;
+		return 0;
```

Comment:
This is my humble opinion, I think it's fine to always return SPI_processed. The return value is simpler.

Reply:
Hmm, I'd like consider the consistency of returned result more than this simplicity. However, the relation between the returned value and recalc_tuptable might have been unclear, so I'd changed it to return the number of rows via a pointer argument.